### PR TITLE
RD-2807 Tests for number of blueprints widget

### PR DESCRIPTION
**Pull request by leon-good-life**
## Description
Tests for number of blueprints widget.
I took inspiration from `/cypress/integration/widgets/deployment_num_spec.ts`.
It includes two tests:
- displays the correct number of resources
- opens chosen page after configuration change on click
- opens the default page

## Screenshots / Videos

Updated:
https://user-images.githubusercontent.com/17104212/140027565-f4ec65f7-99ed-48e0-b5e6-28048c8b4d38.mov

## Checklist

- [x] My code follows the UI Code Style.
- [x] I verified that all tests and checks have passed.
- [x] I verified if that change requires a documentation update.
- [x] I added proper labels to this PR.

## Tests
[Link to the system tests in Jenkins](https://jenkins.cloudify.co/blue/organizations/jenkins/Stage-UI-System-Test/detail/Stage-UI-System-Test/1476/pipeline)
[Link to tests pipelines in Jenkins](https://jenkins.cloudify.co/blue/organizations/jenkins/cloudify-stage/detail/RD-2807/10/pipeline)

 

## Documentation
[Link to the Jira ticket (RD-2807)](https://cloudifysource.atlassian.net/browse/RD-2807)

### CODE REVIEW

**Comment by qooban** [comment]
I just did quick scanning and found a typo `setMoltipleDropdownValues`, should be `setMultipleDropdownValues` :-)

Will review properly tomorrow.

File: test/cypress/integration/widgets/blueprint_num_spec.ts [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    }
+
+    describe('Tests without mockLogin', () => {
+        before(() => cy.activate().login().addWidget(widgetId));
+
+        it('Opens the default page on click', () => {
+            clickOnWidget();
+            verifyUrl('blueprints');
+        });
+    });
```
Comment (qooban) [inline test/cypress/integration/widgets/blueprint_num_spec.ts:25]: I think we can merge it as there's no need to have `describe` section when there's just one test case inside IMHO.

```suggestion
    it('opens the default page on click', () => {
        cy.activate().login().addWidget(widgetId);
        clickOnWidget();
        verifyUrl('blueprints');
    });
```

General:

Strings in `describe` and `it` statements are intended to create a sentence when concatenated. See: https://jenkins.cloudify.co/blue/organizations/jenkins/Stage-UI-System-Test/detail/Stage-UI-System-Test/1445/tests . For example "License Management is providing License Page that fetches license on page load" comes from such code:
```
describe('License Management', () => {
    // ...
    describe('is providing License Page that', () => {
        // ...        
        it('fetches license on page load', () => {
            // ...
```

File: test/cypress/integration/widgets/blueprint_num_spec.ts [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+    describe('Tests with mockLogin', () => {
```
Comment (qooban) [inline test/cypress/integration/widgets/blueprint_num_spec.ts:27]: I think we don't need such description here as when concatenated the full name of those tests will be:
1. "Number of Blueprints widget Tests with mockLogin displays the correct number of resources"
2. "Number of Blueprints widget Tests with mockLogin opens chosen page after configuration change on click"

Having
```suggestion
    describe('', () => {
```
will make it more readable.

File: test/cypress/integration/widgets/deployment_num_spec.ts [hunk 3]
```diff
@@ -55,7 +55,7 @@ describe('Number of Deployments widget', () => {
         function setWidgetConfiguration(filterId: string, pageToOpenOnClick: string) {
             cy.editWidgetConfiguration(widgetId, () => {
                 cy.setSearchableDropdownValue('Filter ID', filterId);
-                cy.setDropdownValues('Page to open on click', [pageToOpenOnClick]);
+                cy.setMultipleDropdownValues('Page to open on click', [pageToOpenOnClick]);
```
Comment (qooban) [inline test/cypress/integration/widgets/deployment_num_spec.ts:58]: It shouldn't be `setMultipleDropdownValues`. 

It uses the same page selector as "Number of Blueprints" widget, so we should change it to:
```suggestion
                cy.setSingleDropdownValue('Page to open on click', pageToOpenOnClick);
```

**Review by qooban — COMMENTED** [review]
Added some comments, mostly minor, related to test styling.

Did you tried full system tests run on this branch (https://jenkins.cloudify.co/job/Stage-UI-System-Test/build?delay=0sec)?

**Comment by leon-good-life** [comment]
@qooban  🙏Thanks a lot for all the reviews.
In the future PRs I will make an effort to make the PRs and the commits better in the first place, so they will require less review rounds and will have a clean commit history.

In this PR, I made a mistake when I committed my code, so the commit history doesn’t seem clean to me and I’m not proud of it.

I wonder if I can combine my commits using ‘git rebase --interactive’ (I know it is terrible idea to rewrite history of pushed code) 
Or maybe to archive this PR, delete the branch, and create a new branch and a PR with only one commit. 

Or leave it as is, and just pay more attention to this in the future.

Thanks again.

* I wrote the same comment in another PR that I worked on at the same time.

**Comment by qooban** [comment]
Thanks for your comment. See my reply here: https://github.com/cloudify-cosmo/cloudify-stage/pull/1648#issuecomment-961682086

**Review by qooban — APPROVED** [review]
Looks good 👍🏻 

Please check if all system tests pass and if so, you can merge it.

**Comment by leon-good-life** [comment]
I ran the system tests: [https://jenkins.cloudify.co/blue/organizations/jenkins/Stage-UI-System-Test/detail/Stage-UI-System-Test/1475/pipeline/55/](https://jenkins.cloudify.co/blue/organizations/jenkins/Stage-UI-System-Test/detail/Stage-UI-System-Test/1475/pipeline/55/)
Only one test was failed: pages_edit_spec
I run it on my machine on this branch and the test passes

**Comment by qooban** [comment]
@leon-good-life Please be careful and do not use "Rebase and merge". Git history on `master` branch is polluted right now with all the commits from this PR. Please read again my comment https://github.com/cloudify-cosmo/cloudify-stage/pull/1648#issuecomment-961682086 and use "Squash and merge" strategy.

**Pull request merged**